sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -13,15 +13,99 @@
 # limitations under the License.
 
 import collections
+import os
 import re
 from typing import Dict, List
 
+import soundfile
+from sox import Transformer
+
 from sdp.logging import logger
 from sdp.processors.base_processor import BaseParallelProcessor, DataEntry
 from sdp.utils.edit_spaces import add_start_end_spaces, remove_extra_spaces
 from sdp.utils.get_diff import get_diff_with_subs_grouped
 
 
+class GetAudioDuration(BaseParallelProcessor):
+    """
+    Processor to count audio duration using audio file path from input_field
+
+    Args:
+        audio_filepath_field (str): where to get path to wav file.
+        duration_field (str): where to put to audio duration.
+    Returns:
+        All the same fields as in the input manifest plus output_field
+    """
+
+    def __init__(
+        self,
+        audio_filepath_field: str,
+        duration_field: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.audio_filepath_field = audio_filepath_field
+        self.duration_field = duration_field
+
+    def process_dataset_entry(self, data_entry):
+        audio_filepath = data_entry[self.audio_filepath_field]
+        try:
+            data, samplerate = soundfile.read(audio_filepath)
+            data_entry[self.duration_field] = data.shape[0] / samplerate
+        except Exception as e:
+            logger.warning(str(e) + " file: " + audio_filepath)
+            data_entry[self.duration_field] = -1.0
+        return [DataEntry(data=data_entry)]
+
+
+class SoxConvert(BaseParallelProcessor):
+    """
+    Processor for converting audio files from one format to another using Sox,
+    and updating the dataset with the path to the converted audio files.
+
+    Args:
+        converted_audio_dir (str): Directory to store the converted audio files.
+        input_field (str): Field in the dataset representing the path to input audio files.
+        output_field (str): Field to store the path to the converted audio files in the dataset.
+        key_field (str, optional): Field in the dataset representing the unique key or identifier for each entry.
```

Comment:
we don't use this, please remove this line

Reply:
done